File: book/스프링부트를 활용한 마이크로서비스 개발/test/src/main/java/com/example/test/domain/Multiplication.java
package com.example.test.domain;

import lombok.*;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;

@RequiredArgsConstructor
@Getter
@ToString
@EqualsAndHashCode
@Entity
public final class Multiplication {

    @Id
    @GeneratedValue
    @Column(name = "MULTIPLICATION_ID")
    private Long id;

    private final int factorA;
    private final int factorB;

    protected Multiplication() {
        this(0,0);
    }
}

File: book/스프링부트를 활용한 마이크로서비스 개발/kotlin/src/main/kotlin/com/example/kotlin/domain/Multiplication.kt
package com.example.kotlin.domain

/*
* 애플리케이션에 곱셈을 나타내는 클래스
* */
class Multiplication(
    var factorA: Int,
    var factorB: Int,
    var result: Int = factorA * factorB
) {
    override fun toString(): String {
        return "Multiplication(factorA=$factorA, factorB=$factorB, result(A*B)=$result)"
    }
}